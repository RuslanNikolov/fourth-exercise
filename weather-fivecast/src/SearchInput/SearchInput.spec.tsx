import React from 'react';
import renderer from 'react-test-renderer';
import SeachInput from './SearchInput';
import enzyme from 'enzyme';

it('renders correctly', () => {
    const component = renderer.create(<SeachInput setCurrentCity={() => { }} />).toJSON();
    expect(component).toMatchSnapshot();
});

it('calls the setCurrentCity function from props', () => {
    const spyFunc = jest.fn();
    const component = enzyme.shallow(<SeachInput setCurrentCity={spyFunc} />);
    component.find('.submit-button').at(0).simulate('click');
    expect(spyFunc).toHaveBeenCalled();
})