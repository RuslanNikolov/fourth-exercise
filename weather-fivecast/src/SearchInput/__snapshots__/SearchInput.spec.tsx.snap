// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBox-root css-0"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
>
  <div
    aria-describedby="city-input"
    className="MuiFormControl-root MuiTextField-root SearchInput css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    style={
      Object {
        "marginRight": "20px",
      }
    }
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="SearchInput"
      id="SearchInput-label"
    >
      City
    </label>
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={true}
        className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="SearchInput"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Narnia is not a real place..."
        required={false}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-dhh0px"
        >
          <span>
            City
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root submit-button css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Let's go!
  </button>
</div>
`;
